autorest/azure/async.go

```diff
@@ -86,7 +88,23 @@ func (f Future) PollingMethod() PollingMethodType {
 }
 
 // Done queries the service to see if the operation has completed.
+// Deprecated: Use DoneWithContext()
 func (f *Future) Done(sender autorest.Sender) (bool, error) {
+	return f.DoneWithContext(context.Background(), sender)
+}
+
+// DoneWithContext queries the service to see if the operation has completed.
+func (f *Future) DoneWithContext(ctx context.Context, sender autorest.Sender) (done bool, err error) {
+	ctx = tracing.StartSpan(ctx, "github.com/Azure/go-autorest/autorest/azure/async.DoneWithContext")
+	defer func() {
+		sc := -1
```

Comment:
Why -1, is this the convention?

Reply:
now that you mentioned if we never get a reply from a server, what is going to be our status code. 0 ? if so we should use 0

Reply:
updated to use an http to tracing status function according to the spec. for any different http code it will report unknown so here we can pass anything except the valid supported http codes and it will be fine